src/file_pipe_log.rs

```diff
@@ -596,6 +596,13 @@ impl<B: FileBuilder> FilePipeLog<B> {
                                 f.fd.truncate(reader.valid_offset())?;
                                 break;
                             }
+
+                            Err(e) if recovery_mode == RecoveryMode::TolerateCorruptedAllRecords =>
+                            {
+                                warn!("All corrupted raft logs will be truncated,  error of one file: {}",  e);
+                                f.fd.truncate(reader.valid_offset())?;
+                                break
+                            }
```

Comment:
It's not as simple as that. A corruption here would break the continuum of certain raft groups, and cause panics when memtables of those raft groups are restored. In specific, this assertion: https://github.com/tikv/raft-engine/blob/745e494bf9acafd88703bf3351cf93040d059ebb/src/memtable.rs#L116-L121